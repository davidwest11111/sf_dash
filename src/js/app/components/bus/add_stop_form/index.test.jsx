import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import AddStopForm from './index';


describe('outputs the expected tree when', () => {
  let wrapper;

  test('(default)', () => {
    wrapper = shallow((
      <AddStopForm
        toggleAddStopForm={() => {}}
        onAddOrRemoveStop={() => {}}
      />
    ));
  });

  test('toggle tab2', () => {
    wrapper = shallow((
      <AddStopForm
        toggleAddStopForm={() => {}}
        onAddOrRemoveStop={() => {}}
      />
    ));
    wrapper.instance().toggle('2');
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('instance methods', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow((
      <AddStopForm
        toggleAddStopForm={() => {}}
        onAddOrRemoveStop={() => {}}
      />
    ));
    instance = wrapper.instance();
  });

  test('button clicks call toggle', () => {
    const spy = jest.spyOn(instance, 'toggle');

    wrapper.find('NavLink').at(0).simulate('click');
    expect(spy).toHaveBeenCalledWith('1');

    wrapper.find('NavLink').at(1).simulate('click');
    expect(spy).toHaveBeenCalledWith('2');
  });
});
