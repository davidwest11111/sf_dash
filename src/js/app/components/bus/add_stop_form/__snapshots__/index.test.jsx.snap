// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when (default) 1`] = `
<div
  className="add-form mb-2"
>
  <Nav
    tabs={true}
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="clickable active"
        onClick={[Function]}
        tag="a"
      >
        Geolocation
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className="clickable"
        onClick={[Function]}
        tag="a"
      >
        Manual Input
      </NavLink>
    </NavItem>
    <button
      className="btn close-form"
      onClick={[Function]}
      type="button"
    >
      <span
        className="fas fa-times"
      />
    </button>
  </Nav>
  <TabContent
    activeTab="1"
    tag="div"
  >
    <TabPane
      tabId="1"
      tag="div"
    >
      <GeolocateAdd
        className=""
        onAddOrRemoveStop={[Function]}
      />
    </TabPane>
    <TabPane
      tabId="2"
      tag="div"
    >
      <ManualAdd
        className=""
        onAddOrRemoveStop={[Function]}
      />
    </TabPane>
  </TabContent>
</div>
`;

exports[`outputs the expected tree when toggle tab2 1`] = `
<div
  className="add-form mb-2"
>
  <Nav
    tabs={true}
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="clickable"
        onClick={[Function]}
        tag="a"
      >
        Geolocation
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className="clickable active"
        onClick={[Function]}
        tag="a"
      >
        Manual Input
      </NavLink>
    </NavItem>
    <button
      className="btn close-form"
      onClick={[Function]}
      type="button"
    >
      <span
        className="fas fa-times"
      />
    </button>
  </Nav>
  <TabContent
    activeTab="2"
    tag="div"
  >
    <TabPane
      tabId="1"
      tag="div"
    >
      <GeolocateAdd
        className=""
        onAddOrRemoveStop={[Function]}
      />
    </TabPane>
    <TabPane
      tabId="2"
      tag="div"
    >
      <ManualAdd
        className=""
        onAddOrRemoveStop={[Function]}
      />
    </TabPane>
  </TabContent>
</div>
`;
